radicle-node/src/service.rs

```diff
@@ -860,14 +871,31 @@ where
                         match message.is_fresh(&self.storage) {
                             Ok(is_fresh) => {
                                 if is_fresh {
-                                    // TODO: Only fetch if the refs announced are for peers we're tracking.
-                                    self.fetch(message.rid, announcer);
+                                    let repo_tracking = self
+                                        .tracking
+                                        .repo_entry(&message.rid)
+                                        .expect("handle_announcement: error accessing repo tracking configuration");
+
+                                    if let Some((Scope::Trusted, _policy)) = repo_tracking {
+                                        for (pub_key, _refs) in message.refs.iter() {
+                                            if self.tracking.is_node_tracked(pub_key)
+                                            .expect("handle_announcement: error accessing node tracking configuration") {
+                                                self.fetch(message.rid, Namespaces::One(*pub_key), announcer);
```

Comment:
Maybe we should rebase this on @FintanH's patch, which allows for setting multiple pks in a Namespaces object.

Reply:
Sure, sounds good. Can I rebase after @FintanH's patch got merged in?